File: scala/scala-impl/src/org/jetbrains/plugins/scala/ScalaBundle.scala
package org.jetbrains.plugins.scala

import com.intellij.DynamicBundle
import org.jetbrains.annotations.{Nls, PropertyKey}

import scala.annotation.varargs

object ScalaBundle {
  private final val BUNDLE = "messages.ScalaBundle"
  private object INSTANCE extends DynamicBundle(BUNDLE)

  //noinspection ReferencePassedToNls
  @Nls
  @varargs
  def message(@PropertyKey(resourceBundle = BUNDLE) key: String, params: Any*): String =
    INSTANCE.getMessage(key, params.map(_.toString): _*)
}

File: scala/scala-impl/src/org/jetbrains/plugins/scala/ScalaBundle.java
package org.jetbrains.plugins.scala;

import com.intellij.DynamicBundle;
import org.jetbrains.annotations.Nls;
import org.jetbrains.annotations.NonNls;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.PropertyKey;

public final class ScalaBundle extends DynamicBundle {
    @NonNls
    private static final String BUNDLE = "messages.ScalaBundle";

    private static final ScalaBundle INSTANCE = new ScalaBundle();

    private ScalaBundle() {
        super(BUNDLE);
    }

    @Nls
    public static String message(@NotNull @PropertyKey(resourceBundle = BUNDLE) String key, @NotNull Object... params) {
        return INSTANCE.getMessage(key, params);
    }
}
